packages/@apphosting/adapter-nextjs/src/bin/build.ts

```diff
@@ -4,35 +4,22 @@ import {
   build,
   populateOutputBundleOptions,
   generateOutputDirectory,
-  DEFAULT_COMMAND,
   validateOutputDirectory,
 } from "../utils.js";
 import { join } from "path";
+import { getBuildOptions } from "@apphosting/common";
 
 const root = process.cwd();
-
-let projectRoot = root;
-if (process.env.FIREBASE_APP_DIRECTORY) {
-  projectRoot = join(root, process.env.FIREBASE_APP_DIRECTORY);
-}
-
-// Parse args to pass to the build command
-let cmdArgs: string[] = [];
-if (process.env.MONOREPO_BUILD_ARGS) {
-  cmdArgs = process.env.MONOREPO_BUILD_ARGS.split(",");
-}
-
-// Determine which command to run the build
-const cmd = process.env.MONOREPO_COMMAND || DEFAULT_COMMAND;
+const opts = getBuildOptions();
 
 // Run build command from the subdirectory if specified.
 // N.B. We run the build command from the root for monorepo builds, so that the build process can
 // locate necessary files outside the project directory.
-build(process.env.MONOREPO_CMD ? root : projectRoot, cmd, ...cmdArgs);
+build(root, opts);
```

Comment:
Would it be possible to simply this contract more on the adapter side? (This would probably complicate the getBuildOptions function).

I'm envisioning something like a new function buildMonorepoIfNeeded, that will handle 100% of monorepo related things, including the nx/turbo/... build. That way the adapter authors would need to know less about monorepos. This would help greatly in the case of community adapters where I'd be worried people might find monorepo support confusing.

I'm imagining something like this:

buildMonorepoIfNeeded(...){
  // pull in the env vars the same way getBuildOptions does
  // run the monorepo build if needed
  return didMonorepoBuildRun, projectDirectory 
}
then later to use it:
didMonorepoBuildRun, projectDirectory = buildMonorepoIfNeeded(...)

if !didMonorepoBuildRun{
  build(root, opts)
}
then the rest is the same.

I like this way better since its more contained within a few lines and I think it might be easier for our docs to provide an example of how to use this

Do you think this would be possible and generalizable to other monorepo frameworks too? Also open to other ideas too if you think this can be simplified another way.